### PR TITLE
Remove shipping methods controller

### LINKED ISSUE #9486: [Bye Bye Angular] Enterprise Settings -> 'Shipping Methods'
Implements Stimulus the "Shipping Methods" section:

 Task One: Remove the use of `shipping_methods_controller` 
(see `app/assets/javascripts/admin/shipping_methods/controllers/shipping_methods_controller.js.coffee`)
Estimate: TBC

### PR DESCRIPTION
**Pull request by binarygit**
#### What? Why?

Closes #9486 

#### What should we test?

- Login as Enterpise User / Super Admin
- Visit page `admin/enterprises/<shop name>/edit#!#shipping_methods_panel`
shipping

- Ascertain that the shipping methods allowed by the enterprise are checked
- Ascertain that the shipping methods not allowed by the enterprise are not checked

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by binarygit** [comment]
Both these specs are passing locally :shrug: 

specs
specs

**Comment by sigmundpetersen** [comment]
Yes, one of them is #9643 , we have a build failure in master at the moment. So not related to this PR.
I wonder if the other one is a flaky spec, I will try to rebuild to see if we are left with only the other failure 👍

**Comment by sigmundpetersen** [comment]
Yep, only #9643 after a rebuild.
I say this is good for review.

**Review by jibees — APPROVED** [review]
This looks so simple, almost unreal. Why we created this `shippingMethodsCtrl`?

**Comment by binarygit** [comment]
From what I've read of the code and the corresponding tests, this controller is supposed to:
1. Check only the shipping methods allowed/selected by the enterprise
2. If there are no shipping methods allowed/selected by the enterprise then display <some element> in `red`

The first one is done through Rails itself, I think when this was originally done, the dev writing it didn't know that rails could do it, so they ended up writing it in Angular.
For the second one, I haven't found the <some element> it's supposed to display in `red`. I'm guessing it got removed somewhere along the way.

Also, could you take a look at this too?  https://github.com/openfoodfoundation/openfoodnetwork/pull/9506
It is also doing the same thing as this one :laughing:

**Comment by jibees** [comment]
I'd say this `displayColor()` method is not called anymore. Let's delete it, and see how it goes!

**Comment by binarygit** [comment]
yup I already deleted it and nothing is breaking nor is there any change in the page :slightly_smiling_face:

**Comment by mkllnk** [comment]
I just rebased on master to check the specs again.

File: spec/javascripts/unit/admin/enterprises/services/enterprise_shipping_methods_spec.js.coffee [hunk 1]
```diff
@@ -1,44 +0,0 @@
-describe "EnterpriseShippingMethods service", ->
```
Comment (mkllnk) [inline spec/javascripts/unit/admin/enterprises/services/enterprise_shipping_methods_spec.js.coffee:1]: Did you intend to delete the service as well? `app/assets/javascripts/admin/enterprises/services/enterprise_shipping_methods.js.coffee`

**Review by mkllnk — COMMENTED** [review]
Great find. I find it so satisfying to delete unused code.

Just one question below.

**Comment by jibees** [comment]
> I just rebased on master to check the specs again.

🔴 Specs failed, and I think it's relevant. Back to `In Dev` then.

File: spec/javascripts/unit/admin/enterprises/services/enterprise_shipping_methods_spec.js.coffee (hunk 1 above)
Comment (binarygit) [reply spec/javascripts/unit/admin/enterprises/services/enterprise_shipping_methods_spec.js.coffee:1]: I just pushed smoothing out all the broken tests. Could you take a look please?

**Review by mkllnk — APPROVED** [review]
Looks good to me.

**Review by jibees — APPROVED** [review]
Looks good to me!

**Comment by drummer83** [comment]
Hi @binarygit,
Thanks for yet another PR! :fire: 

I have tested as enterprise user as well as super admin user:
- create new shipping method
- add existing shipping method to the hub
- remove existing shipping method from the hub
- edit shipping method

All shipping methods that were active before the PR are still active after the PR. :heavy_check_mark: 
All shipping methods that were inactive before the PR are still inactive after the PR. :heavy_check_mark: 

**Before PR**
image

**After PR**
image

I couldn't find anything special. We're ready to go here! :rocket:

**Pull request merged**